rero_ils/modules/organisations/api.py

```diff
@@ -136,21 +127,48 @@ def online_circulation_category(self):
         except StopIteration:
             return None
 
-    @classmethod
-    def get_record_by_online_harvested_source(cls, source):
-        """Get record by online harvested source."""
-        results = OrganisationsSearch().filter(
-            'term', online_harvested_source=source).scan()
-        try:
-            return Organisation.get_record_by_pid(next(results).pid)
-        except StopIteration:
-            return None
-
     def get_online_locations(self):
         """Get list of online locations."""
         return [library.online_location
                 for library in self.get_libraries() if library.online_location]
 
+    def get_libraries_pids(self):
+        """Get all libraries pids related to the organisation."""
+        results = LibrariesSearch().source(['pid'])\
+            .filter('term', organisation__pid=self.pid)\
+            .scan()
+        for result in results:
+            yield result.pid
+
+    def get_libraries(self):
+        """Get all libraries related to the organisation."""
+        pids = self.get_libraries_pids()
+        for pid in pids:
+            yield Library.get_record_by_pid(pid)
+
+    def get_number_of_libraries(self):
+        """Get number of libraries."""
+        return LibrariesSearch()\
+            .filter('term', organisation__pid=self.pid)\
+            .source()\
+            .count()
+
+    def get_links_to_me(self):
+        """Get number of links."""
+        links = {}
+        libraries = self.get_number_of_libraries()
+        if libraries:
+            links['libraries'] = libraries
```

Comment:
`links` is useless this can be rewritten as:
```python
libraries = self.get_number_of_libraries()
if libraries:
    return dict(libraries=libraries)
```

Reply:
In all cases, to not break the process, I need to return a dict (even i empty).
So the current solution, with only one return statement, seems as good as your proposal.